api/src/domain/user/encryptFieldsFactory.ts

```diff
@@ -1,21 +1,24 @@
-import { EncryptionDecryptionClient } from "@domain/types";
+import { CryptoClient } from "@domain/types";
 import { maskTaxId } from "@domain/user/maskTaxId";
 import { getCurrentBusiness } from "@shared/domain-logic/getCurrentBusiness";
 import { modifyCurrentBusiness } from "@shared/domain-logic/modifyCurrentBusiness";
 import { maskingCharacter } from "@shared/profileData";
 import { UserData } from "@shared/userData";
 
 export const encryptFieldsFactory = (
-  encryptionDecryptionClient: EncryptionDecryptionClient,
+  encryptionDecryptionClient: CryptoClient,
+  hashingClient: CryptoClient,
 ): ((userData: UserData) => Promise<UserData>) => {
   return async (userData: UserData): Promise<UserData> => {
+    const userDataWithHashedTaxId = await hashProfileTaxId(userData, hashingClient);
```

Comment:
Added this line ahead of the other `encryptionFunctions` for 2 reasons:
1. The hashing needs to occur before encryption because after encryption, we're left with a masked value such as `********1234`. We do not want to hash the masked value.
2. We're using 2 different `CryptoClient`s. We could have restructured this logic to tie the function to the associated CryptoClient, but that would have made this a bit harder to read.